pkg/controller/statusmanager/status_manager.go

```diff
@@ -48,20 +52,77 @@ const (
 type StatusManager struct {
 	sync.Mutex
 
-	client crclient.Client
-	mapper meta.RESTMapper
+	client cnoclient.Client
 	name   string
 
 	failing         [maxStatusLevel]*operv1.OperatorCondition
 	installComplete bool
 
 	daemonSets     []types.NamespacedName
 	deployments    []types.NamespacedName
+	statefulSets   []types.NamespacedName
 	relatedObjects []configv1.ObjectReference
+
+	hypershiftConfig network.HypershiftConfig
+}
+
+func New(client cnoclient.Client, name string) *StatusManager {
+	return &StatusManager{
+		client:           client,
+		name:             name,
+		hypershiftConfig: network.NewHypershiftConfig(),
+	}
+}
+
+// setCOAnnotation sets an annotation on the clusterOperator network object
+func (status *StatusManager) setCOAnnotation(obj *configv1.ClusterOperator) error {
```

Comment:
Can we rename this to `setClusterOperAnnotation` or similar? CO is a little confusing for me

Reply:
Done